corehq/apps/users/models.py

```diff
@@ -1488,6 +1487,9 @@ def save(self, fire_signals=True, update_django_user=True, **params):
                 self._save_user_data()
             super(CouchUser, self).save(**params)
 
+        # wait to clear cache until the user has been successfully saved to couch
+        # otherwise a method could trigger caching the previous value of the user despite the save
+        self.clear_quickcache_for_user()
```

Comment:
The only suggestion I'd make is to put this isn a `finally` block with the `save` above in the `try`. That way if there's some stuff executed on the signal that fails, we still always clear the cache.

Reply:
Similar to https://github.com/dimagi/commcare-hq/blob/2dcf4d479c3a1749c7be6ed72c32ad440f139e9e/corehq/apps/cachehq/mixins.py#L18-L22

Reply:
Added in 2449a704e8cae240fb4498aa94e1e528a8d5ec2b